WapTenantPublicAPI.psm1

```diff
@@ -2742,6 +2909,68 @@ function New-WAPVM {
     }
 }
 
+function Set-WAPVM {
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Mandatory,
+            ValueFromPipeline)]
+        [ValidateNotNull()]
+        [PSCustomObject] $VM,
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [int32] $CPU,
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [int32] $Memory,
```

Comment:
Cannot be a negative number so please use uint. No need for ValidateNotNullOrEmtpy